src/Http/Cookie/CookieInterface.php

```diff
@@ -49,13 +49,13 @@ public function getName(): string;
     public function getValue();
 
     /**
-     * Gets the cookie value as a string.
+     * Gets the cookie value as scalar.
      *
      * This will collapse any complex data in the cookie with json_encode()
      *
-     * @return string
+     * @return mixed
      */
-    public function getStringValue();
+    public function getScalarValue();
```

Comment:
Do we need to document this in the migration guide etc?

Reply:
The old method still exists with deprecated tag but yeah mentioning it in migration guide would also be nice.